pkg/services/ngalert/state/historian/loki_http.go

```diff
@@ -297,3 +286,36 @@ type QueryRes struct {
 type QueryData struct {
 	Result []Stream `json:"result"`
 }
+
+func (c *HttpLokiClient) handleLokiResponse(res *http.Response) ([]byte, error) {
+	if res == nil {
+		return nil, fmt.Errorf("response is nil")
+	}
+
+	data, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading request response: %w", err)
+	}
+
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
+		if len(data) > 0 {
+			c.log.Error("Error response from Loki", "response", string(data), "status", res.StatusCode)
+		} else {
+			c.log.Error("Error response from Loki with an empty body", "status", res.StatusCode)
+		}
+		return nil, fmt.Errorf("received a non-200 response from loki, status: %d", res.StatusCode)
+	}
+
+	return data, nil
+}
+
+// ClampRange ensures that the time range is within the configured maximum query length.
+func ClampRange(start, end, maxTimeRange int64) (newStart int64, newEnd int64) {
```

Comment:
nit: could we add a unit test in this package for this? Right now the logic is here but the tests live elsewhere.

Reply:
Ah yeah good call 👍